### ISSUE TITLE
Background geolocation uses less accurate method(s), not GPS

### ISSUE THREAD

**Issue by RikdeBoer** [pmwisdom/cordova-background-geolocation-services#49]
Hello,

Still having great fun with this plugin. I was wondering whether anyone has similar experiences to the symptoms described below and what they've done to fix them.

On a test drive I noticed that on my Android (v 5), in background mode, the position updates came in sporadically and inaccurately, especially when driving in the country side.
The _foreground_ updates came in every second and on average have an accuracy of 3..10 meters, both in the city and in the country.
In the country, background updates were irregular and far and few between (maybe once, twice per minute). The accuracy of the updates was never better than 45 meters and sometimes huge (1000 meters), even though I specified desiredAccuracy=50. 
It was much better in the city, but even there the location updates did not appear to obey the parameter settings passed into the plugin.

Then I realised that in background mode, the plugin must not be using GPS. It's probably using Wifi etc, which explains the differences seen between country and city behaviour. At first I thought the absence of GPS might be due to "battery saving mode", but it happens both with nearly full and nearly empty batteries and regardless of whether the phone is plugged into a charger (cigarette lighter) or not. 

Everything changed for the better when I installed this plugin: https://play.google.com/store/apps/details?id=com.webmajstr.gpson&hl=e

Now, using this plugin, my background position updates on Android have the same accuracy as the foreground position updates (3..10 m) and the update frequency appears to obey the settings I gave it (interval=4000 ms in my case).

So with the above GPS Keeper app I now have a workaround in place. And I'm pretty happy.

But am still wondering -- is this how it is meant to work? 
Have I misconfigured the plugin?
Is it the version of Android? 
I haven't built to iOS yet, so cannot offer any info on that front.

**Comment by pmwisdom** [comment]
1. It is using the GPS. 
2. You requested desired accuracy = 50 so you won't get many points with "lower" accuracy than that.  The lower accuracy,  the more accurate, as the comments say.

**Comment by RikdeBoer** [comment]
I have some more evidence that in background many Android devices do **NOT use GPS**, unless you use an app like GPS Keeper Lite. https://play.google.com/store/apps/details?id=com.webmajstr.gpson&hl=e

I'm not saying this is due to this wonderful plugin, I'm just sharing what I and a friend have observed. 

I wrote a little test app, using this plugin. It records the background location updates and displays them on a map when you return to the foreground. 
Below is a screenshot of my journey on a tram, in Melbourne, Australia.
![backgroundgpsvswifi](https://cloud.githubusercontent.com/assets/4031225/17827926/448fe6e6-66c9-11e6-94c7-045e5f02c963.png)
Each circle represents a background position update, with the radius of the circle equal to the accuracy as reported by the plugin. Coming from the South, it is clear where I turned the GPS Keeper App on: suddenly the circles are smaller (higher accuracy). 
Mind you, this is in the city, the results are much more dramatic (bigger circles, appearing few and far between) when out of town, where there's little WiFi/cellular around.

This was with my mobile location configuration set to _"High accuracy, use GPS, WiFi and cellular networks"_. 

I am convinced that most Androids insist on using Wifi/cellular (not GPS) when in background mode, while GPS is used in foreground mode (if so configured).

If you try your Android to force to use GPS in the background by selecting "_Device only, use GPS"_, and you don't use GPS Keeper, then you will get this error from the plugin: "Location Provider is not available. Maybe GPS is disabled or the provider was rejected?"

In summary, use GPS Keeper and all will be fine.

![moblocsettings](https://cloud.githubusercontent.com/assets/4031225/17827933/5fbf9218-66c9-11e6-928b-aacb2ca57738.png)

**Comment by diegobill** [comment]
Same problem, any solution built-in (solution alternative to download a app like GPS Keeper Lite) to cordova ?

**Comment by rimvydasr** [comment]
To get accurate position in background use   desiredAccuracy: 0,  configuration option because internal plugin code uses LocationRequest.PRIORITY_BALANCED_POWER_ACCURACY instead of LocationRequest.PRIORITY_HIGH_ACCURACY if desiredAccuracy is above zero.

from BackgroundLocationUpdateService.java:

` private Integer translateDesiredAccuracy(Integer accuracy) {`
`        if(accuracy <= 0) {`
      `      accuracy = LocationRequest.PRIORITY_HIGH_ACCURACY;`
   `     } else if(accuracy <= 100) {`
         `   accuracy = LocationRequest.PRIORITY_BALANCED_POWER_ACCURACY;`
 `       } else if(accuracy  <= 1000) {`
      `      accuracy = LocationRequest.PRIORITY_LOW_POWER;`
      `  } else if(accuracy <= 10000) {`
           ` accuracy = LocationRequest.PRIORITY_NO_POWER;`
      `  } else {`
     `     accuracy = LocationRequest.PRIORITY_BALANCED_POWER_ACCURACY;`
  `      }`
      `  return accuracy;`
 `   }`
